### ISSUE TITLE
Split each screen of multi-screen on different ports

### ISSUE THREAD

**Issue by Elko29** [TurboVNC/turbovnc#141]
Hi everybody,
I'm looking for a solution to separate the broadcasting of each screen from a mutli-screen on different ports.

Exemple, with these command line : 
```
                vncserver -geometry 1920x1080+0+0,1920x1080+1920+0
                                    \-----------/ \---------------/
What I would like to have :          port : 5901     port : 5902
```

My goal is to have one viewer listening to 5901 for screen 1 and a second one listening to 5902 for screen 2, is it possible or there is another solution to do this ?

**Comment by dcommander** [comment]
No, it isn't currently possible, and I can think of many reasons why implementing such a solution would be problematic at best.  But why do you need such a solution?  I feel like whatever problem you're trying to solve can probably already be solved in a different way.

**Comment by Elko29** [comment]
Thanks !
Today, when I want to do something like my previous question, I need to use multiple turbovnc server on 5901, 5092, ... and aggregate every body with Xdmx and Xinerama tool.
It's functional but very heavy and Xdmx is an old tool ...

I also tried to use Xvfb to generate the screens I need and send this screens with multiple x11vnc but it was very very heavy ....

Have you another solution ?

**Comment by dcommander** [comment]
Version 2.2 (which is currently in beta) of the TurboVNC Server already has built-in Xinerama support, which interacts seamlessly with the multi-screen spanning feature in the TurboVNC Viewer.  I do not understand why you need to attach a different RFB server/port to each screen.  You still have not explained why you feel that you need to do that.  Without understanding the ultimate purpose, I cannot tell you what the appropriate solution would be.

**Comment by Elko29** [comment]
My use case is a wall screen where each screen is a simple vnc viewer. The visual comes from a VNC server that splits the VNC flow.

**Comment by dcommander** [comment]
OK, but why can't you connect all of the screens in the display wall to the same X server?  If you do that, then you should be able to use the TurboVNC Viewer with the new Xinerama feature in the TurboVNC Server to drive the wall.  There are potential optimizations that might need to be made to improve performance, such as multi-threaded decoding in the viewer (see #60), but in general, that would be a much cleaner solution than using a different RFB connection for each screen.

**Comment by dcommander** [comment]
Would it be possible to start multiple TurboVNC Server instances and run Xdmx on the server instead of the client?  Barring that, if the screens are all connected to independent machines, then I don't know of a solution other than what you propose-- but I don't think any VNC servers can currently do that.  Modifying the TurboVNC Server to support such a feature would be tricky and time consuming.  It's unfortunately not a project I could do for free.  Leaving this open and tagging as "funding needed" in case an organization might be interested in paying me to implement the feature.

**Comment by dcommander** [comment]
NOTE: I was able to successfully run Xdmx across multiple TurboVNC Server instances.  I couldn't make it treat each instance as an independent screen, but you probably don't need that for a display wall.
